lib/src/rules/always_specify_types.dart

```diff
@@ -99,10 +100,18 @@ class AlwaysSpecifyTypes extends LintRule implements NodeLintRule {
     registry.addTypeName(this, visitor);
     registry.addVariableDeclarationList(this, visitor);
   }
+
+  void _reportLintForTokenWithDescription(Token token, String description) {
+    reporter.reportErrorForToken(LintCode(name, description), token);
```

Comment:
I believe that the better implementation would be to follow the pattern used in `lib/src/rules/always_declare_return_types.dart`. Doing so
- removes the need for special reporting methods and
- allows the code to have a correction message, which is the better place to put this information.